WordPress/src/main/java/org/wordpress/android/ui/reader/views/ReaderWebView.java

```diff
@@ -188,6 +207,34 @@ public boolean shouldOverrideUrlLoading(WebView view, String url) {
                 return false;
             }
         }
+
+        @Override
+        public WebResourceResponse shouldInterceptRequest(WebView view, String url) {
+            // Intercept requests and add the WP.com authorization header
+            if (!TextUtils.isEmpty(mToken) && isImageUrl(url)) {
+                DefaultHttpClient client = new DefaultHttpClient();
+                HttpGet httpGet = new HttpGet(url);
+                httpGet.setHeader("Authorization", "Bearer " + mToken);
+                try {
+                    HttpResponse httpResponse = client.execute(httpGet);
+                    InputStream responseInputStream = httpResponse.getEntity().getContent();
+                    return new WebResourceResponse(httpResponse.getEntity().getContentType().toString(), "UTF-8", responseInputStream);
+                } catch (IOException e) {
+                    AppLog.e(AppLog.T.READER, "Invalid reader detail request: " + e.getMessage());
+                }
+            }
+
+            return super.shouldInterceptRequest(view, url);
+        }
+    }
+
+    private static boolean isImageUrl(String url) {
+        if (TextUtils.isEmpty(url)) return false;
+
+        String lowerCaseUrl = url.toLowerCase();
+
+        return lowerCaseUrl.endsWith("jpg") || url.endsWith("jpeg") || url.endsWith("gif") ||
```

Comment:
Only the first comparison is using `lowerCaseUrl` - the rest incorrectly use `url`.